### PR TITLE
Update RSyslog HLD with new functionality

### PR DESCRIPTION
**Pull request by iavraham**
Adding the following functionality:

- Configure remote syslog servers: protocol, filter, trap severity level
- Update global syslog configuration: trap severity kevel, message format

these are the implementation PRs for community review:
| Repo  | PR title | State |
| ---------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------- |
| sonic-swss-common | [[rsyslog]add syslog global config to db](https://github.com/sonic-net/sonic-swss-common/pull/771) | GitHub issue/pull request detail |
| sonic-host-services | [[rsyslog]Add remote syslog configuration](https://github.com/sonic-net/sonic-host-services/pull/53)  | GitHub issue/pull request detail |  
| sonic-buildimage | [[rsyslog]Add remote syslog configuration](https://github.com/sonic-net/sonic-buildimage/pull/14513) | GitHub issue/pull request detail |
| sonic-utilities | [[syslog] Adjust runningconfiguration syslog command](https://github.com/sonic-net/sonic-utilities/pull/2843) | GitHub issue/pull request detail |

### CODE REVIEW

File: doc/syslog/syslog-design.md
Comment (fastiuk) [inline doc/syslog/syslog-design.md:9]: We need to get rid of section numbers here and remove them from table of content.
You can use ""  directive. See [here](https://github.com/sonic-net/SONiC/pull/1211)

File: doc/syslog/syslog-design.md
Comment (venkatmahalingam) [inline doc/syslog/syslog-design.md:821]: Can we add a default field?

**Comment by zhangyanzhao** [comment]
@dharmaraj-gurusamy wants to be the reviewer.

File: doc/syslog/syslog-design.md
Comment (dharmaraj-gurusamy) [inline doc/syslog/syslog-design.md:190]: severity will be a better name as trap may confuse with SNMP terminology. Please keep uniform name for both global & server specific configuration parameters.

File: doc/syslog/syslog-design.md
Comment (dharmaraj-gurusamy) [inline doc/syslog/syslog-design.md:260]: Please correct the typo.

File: doc/syslog/syslog-design.md
Comment (dharmaraj-gurusamy) [inline doc/syslog/syslog-design.md:256]: Please share more details on this statement, preferably with an example.

File: doc/syslog/syslog-design.md
Comment (dharmaraj-gurusamy) [inline doc/syslog/syslog-design.md:277]: Can this be renamed as filter_type?

File: doc/syslog/syslog-design.md
Comment (dharmaraj-gurusamy) [inline doc/syslog/syslog-design.md:376]: Is it possible to configure both include & exclude regex simultaneously? Please capture the restrictions in the design.

File: doc/syslog/syslog-design.md
Comment (dharmaraj-gurusamy) [inline doc/syslog/syslog-design.md:643]: Please consider the command options as "--filter_type <include/exclude> --filter_regex <string> --severity <>".

File: doc/syslog/syslog-design.md
Comment (dharmaraj-gurusamy) [inline doc/syslog/syslog-design.md:662]: The extended parameters shall be displayed in show command output also for consistency.

File: doc/syslog/syslog-design.md
Comment (dharmaraj-gurusamy) [inline doc/syslog/syslog-design.md:788]: filter_type will be the better name.

File: doc/syslog/syslog-design.md
Comment (dharmaraj-gurusamy) [inline doc/syslog/syslog-design.md:803]: severity will be the better name.

File: doc/syslog/syslog-design.md
Comment (dharmaraj-gurusamy) [inline doc/syslog/syslog-design.md:831]: Please add the default value.

File: doc/syslog/syslog-design.md
Comment (dharmaraj-gurusamy) [inline doc/syslog/syslog-design.md:209]: Will the default values be added in redis during upgrade scenarios (db_migrator script)? If not, we might end up in inconsistency after upgrade?

**Review by dharmaraj-gurusamy — CHANGES_REQUESTED** [review]

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:190]: we defined this name in our internal usage so we would prefer to leave it like that.
is it acceptable by you?

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:277]: will change

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:788]: will change

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:803]: answered above

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:209]: if the keys are empty we will use the global default value in the template

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:760]: done

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:821]: added

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:1]: fixed

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:74]: added  detailed description

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:256]: added

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:376]: not possible, added restriction in the document.

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:643]: done

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:662]: added

File: doc/syslog/syslog-design.md
Comment (iavraham) [reply doc/syslog/syslog-design.md:831]: added

**Review by dharmaraj-gurusamy — APPROVED** [review]

**Comment by iavraham** [comment]
to merge the submodules, we first need to merge sonic-swss-common and then we can pass CI tests

**Comment by fastiuk** [comment]
> @iavraham any time line when feature can be approved and merged? Thank you!

We answered/closed all comments in this PR. Now we are waiting for maintainers to approve and merge it.

**Comment by wen587** [comment]
Will review before 5/25

File: doc/syslog/syslog-design.md
Comment (wen587) [inline doc/syslog/syslog-design.md:295]: Current 'show runningconfiguration syslog' is matching lines starting with '*.*'. 
Will it break the show cli function? https://github.com/sonic-net/sonic-utilities/blob/master/show/main.py#L1687

Comment (fastiuk) [reply doc/syslog/syslog-design.md:295]: > Will it break the show cli function?

It will. I fixed it here: https://github.com/sonic-net/sonic-utilities/pull/2843

**Review by wen587 — APPROVED** [review]

**Review by venkatmahalingam — APPROVED** [review]

**Review by venkatmahalingam — APPROVED** [review]

**Pull request merged**
